probe_scraper/parsers/pings.py

```diff
@@ -25,18 +27,21 @@ class GleanPingsParser:
     to parse the pings.yaml files.
     """
 
-    def parse(self, filenames, config):
+    def parse(self, filenames, config, repo_url=None, commit_hash=None):
         config = config.copy()
         paths = [Path(fname) for fname in filenames]
         paths = [path for path in paths if path.is_file()]
         results = parse_objects(paths, config)
         errors = [err for err in results]
 
-        return (
-            {
-                normalize_ping_name(ping_name): ping_data.serialize()
-                for category, pings in results.value.items()
-                for ping_name, ping_data in pings.items()
-            },
-            errors,
-        )
+        pings = {
+            normalize_ping_name(ping_name): ping_data.serialize()
```

Comment:
I would prefer the below addition live inside a separate function alongside this serialized call:
```suggestion
            normalize_ping_name(ping_name): generate_definition(ping_data, repo_url, commit_hash)
```

Reply:
Hi @fbertsch could you help elaborate more on this? Since this function only takes `defined_in` and returns the source URL, I think what you recommended might not work. Maybe I'm still not understanding it quite right?

Reply:
No worries - what I'm proposing is a new function inside the `pings.py` file called `generate_definition`. That function would _call_ `get_source_url`, e.g.:

```
def generate_definition(ping_data, repo_url, commit_hash):
  serialized = ping_data.serialize()
  if repo_url and commit_hash:
    serialized["url"] = get_source_url(...)
  return serialized
```

Reply:
That way we only need to iterate over the values once, and it makes it obvious where any future changes to the definitions should live.

Reply:
Ah this made a lot of sense! Thank you for explaining!